cumulusci/tasks/salesforce/tests/util.py

```diff
@@ -27,3 +33,43 @@ def create_task(task_class, options=None, project_config=None, org_config=None):
         return_value="ccitests",
     ):
         return task_class(project_config, task_config, org_config)
+
+
+def patch_dir(patch_path, file_path):
+    directory = Path(file_path)
+    directory.mkdir(parents=True, exist_ok=True)
+    patch = mock.patch(patch_path, file_path)
+    patch.start()
+    return patch
+
+
+@pytest.fixture(scope="function")
+def create_task_fixture(request):
+    "create_task fixture which does support orginfo cache."
+    temp_dirs = TemporaryDirectory()
+    temp_root = Path(temp_dirs.name)
+
+    to_patch = {
+        "cumulusci.core.config.UniversalConfig.cumulusci_config_dir": temp_root
+        / "fixture_userhome/.cumulusci",
+        "cumulusci.tests.util.DummyKeychain.global_config_dir": temp_root
+        / "fixture_userhome/.cumulusci",
+        "cumulusci.core.config.project_config.BaseProjectConfig.project_cache_dir": temp_root
+        / "fixture_userhome/project_home/.cci",
+        "cumulusci.tests.util.DummyKeychain.project_cache_dir": temp_root
+        / "fixture_userhome/project_home/.cci",
+    }
```

Comment:
Is it really necessary to patch all these? I think most would follow naturally if we patch Path.home and make sure the project config's repo_root is set? With this approach I'm afraid we'll be forever tracking down things that didn't get patched and are accidentally writing to the user's real homedir...

Reply:
I pared it back somewhat but DummyKeychain doesn't inherit from real keychain and it broke things when I tried to make it do so.

Reply:
DummyKeychain could implement global_config_dir and project_cache_dir as properties that make use of Path.home()

Reply:
7b98b4fd